docs/AuthenticationMethods.md

```diff
@@ -0,0 +1,107 @@
+#Authentication Methods
+Kusto spark connector allows user to authenticate with AAD to the Kusto cluster using either an AAD application,
+ a user token or device authentication. The application parameters can also be passed using azure keyVault.
+
+## AAD application authentication
+This is the easiest way to authenticate using this connector and is explained used in all the documentation examples.
+
+ * **KUSTO_AAD_CLIENT_ID**: 
+  AAD application (client) identifier.
+  
+ * **KUSTO_AAD_AUTHORITY_ID**: 
+  AAD authentication authority. This is the AAD Directory (tenant) ID.
+ 
+ * **KUSTO_AAD_CLIENT_PASSWORD**: 
+ AAD application key for the client.
+ 
+#### Example
+```
+df.write
+  .format("com.microsoft.kusto.spark.datasource")
+  .partitionBy("value")
+  .option(KustoOptions.KUSTO_CLUSTER, "MyCluster")
+  .option(KustoOptions.KUSTO_DATABASE, "MyDatabase")
+  .option(KustoOptions.KUSTO_TABLE, "MyTable")
+  .option(KustoOptions.KUSTO_AAD_CLIENT_ID, "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx")
+  .option(KustoOptions.KUSTO_AAD_CLIENT_PASSWORD, "MyPassword") 
+  .option(KustoOptions.KUSTO_AAD_AUTHORITY_ID, "AAD Authority Id") // "microsoft.com"
+  .save()
+```
+## KeyVault
```

Comment:
Move before or after the authentication parameters (such as appId/appKey etc.), since this is an alternative method for all these parameters